### PR TITLE
[4] Saving User Profile logs you out of the backend

### PR DESCRIPTION
**Pull request by alikon**
Pull Request for Issue #42151 .

### Testing Instructions

- In Global Configuration -> System, "Shared Session" must be disabled to reproduce the issue
- Create a Menu Item of type "Users » Edit User Profile", call it "Edit Profile"
- Keep the backend open in a browser tab
- Open another tab, go to the frontend, click on the "Edit Profile" Menu Item
- Simply save your profile from the frontend by pressing "Save"
- Go back to the backend tab, refresh. You've been logged out of the backend.
 

### Actual result BEFORE applying this Pull Request
You are logged out from the backend by just editing your profile from the frontend.

### Expected result AFTER applying this Pull Request

not logged out

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [ ] No documentation changes for docs.joomla.org needed

- [ ] Pull Request link for manual.joomla.org: <link>
- [ ] No documentation changes for manual.joomla.org needed

### CODE REVIEW

**Comment by fgsw** [comment]
I have tested this item :white_check_mark: successfully on f61cbe9ecfb6bc765625079a7b1f1fe336a0d325<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/42284">issues.joomla.org/tracker/joomla-cms/42284</a>.</sub>

**Comment by AndySDH** [comment]
I have tested this item :white_check_mark: successfully on f61cbe9ecfb6bc765625079a7b1f1fe336a0d325<br /><br />Thanks<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/42284">issues.joomla.org/tracker/joomla-cms/42284</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/42284">issues.joomla.org/tracker/joomla-cms/42284</a>.</sub>

**Comment by HLeithner** [comment]
I'm a bit surprised why the `$data['password']` is not that same as `$data['password1']` at the if statement, I see no change for this variable between the assignment of `$data['password']` at line 250 and the if statement.

**Comment by HLeithner** [comment]
I looked at the wrong class, bind is used on the User Object and not on the user table which is not a reference.... so it's ok from me

File: components/com_users/src/Model/ProfileModel.php [hunk 1]
```diff
@@ -281,7 +281,7 @@ public function save($data)
         }
 
         // Destroy all active sessions for the user after changing the password
-        if ($data['password']) {
+        if ($data['password1']) {
```
Comment (laoneo) [inline components/com_users/src/Model/ProfileModel.php:284]: I'm just wondering if the condition should not be more like:
`if ($user->id !== $this->getCurrentUser()->id)`

Comment (alikon) [reply components/com_users/src/Model/ProfileModel.php:284]: i don't think so
`$user->id === $this->getCurrentUser()->id` 
we are not changing the id, we need to know if the psw is changed

Comment (laoneo) [reply components/com_users/src/Model/ProfileModel.php:284]: With this check we can test if the current user does edit his own profile. If not, we log him out.

Comment (laoneo) [reply components/com_users/src/Model/ProfileModel.php:284]: Anyway, we go now with it.

**Comment by laoneo** [comment]
Thanks!

**Pull request merged**